atlp-rwanda/e-commerce-bitcrafters-bn
# [#187364664] - Feature(MoMo API): Integrating Mobile money in our payment mechanism

## Pull Request: Mobile Money Payment Integration with MoMo API

### 1. What does this PR do?
This PR introduces a robust integration of the MoMo API into our e-commerce platform, enabling seamless mobile money transactions. It establishes two primary endpoints for initiating payments and verifying transaction status, thereby enhancing the checkout experience for our users.

### 2. Description of Task to be completed?
- Implement the `payment/momo/pay/{orderId}` endpoint to initiate a payment request for an order.
- Develop the `payment/momo/check/{orderId}` endpoint to verify the payment status of an order.
- Ensure secure transmission of data by adhering to the MoMo API security standards.
- Create a seamless workflow that aligns with the existing checkout process.

### 3. How should this be manually tested?
- Navigate to the Swagger documentation of our API.
- Use the `payment/momo/pay/{orderId}` endpoint to simulate a payment process with a valid `orderId`.
- Follow up with the `payment/momo/check/{orderId}` endpoint to check the payment status.
- Verify that the responses align with the expected outcomes and that the transaction reflects accurately in the system.

### 4. Any background context you want to provide?
The integration with MoMo API is part of our strategic move to diversify our payment options and cater to the growing demand for mobile money services. This feature is pivotal in enhancing user convenience and trust, as mobile money is a preferred payment method for many of our customers in the region.


#### Loom video recording
https://www.loom.com/share/cf43e7ba71454f5db3c105b99b059d35?sid=f89acd6f-01a1-4dca-ae92-ff2002a2328b

#### What are the relevant pivotal tracker/Trello stories?
[Delivers #187364664]
